tests/ui/useless_conversion.rs

```diff
@@ -131,6 +131,13 @@ fn main() {
     dont_lint_into_iter_on_copy_iter();
     dont_lint_into_iter_on_static_copy_iter();
 
+    #[allow(clippy::into_iter_on_ref)]
+    {
+        // Should suggest `*items` instead of `&**items`
+        let items = &&[1, 2, 3];
+        let _ = items.into_iter();
```

Comment:
This doesn't actually trigger the lint. You need to pass the result of the call into a function that takes `T: IntoIterator` e.g.
```rust
fn f(_: impl IntoIterator) {}
let x = &&[1];
f(x.into_iter());
```